packages/ui-components/src/Input/Field.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { FunctionComponent } from "react";
+import { Field, FieldProps, FieldRenderProps } from "react-final-form";
+
+import { TextInput, TextProps} from "./TextTypeInput";
+import { EmailInput, NewPasswordInput, ExistingPasswordInput, EmailNumberProps, NewPasswordProps, ExistingPasswordProps } from "./EmailPasswordInput";
+import { CheckboxInput, CheckboxInputProps } from "./CheckboxInput";
+
+type TextInputFieldProps = Omit<FieldProps<
+  string,
+  FieldRenderProps<string, HTMLElement>,
+  HTMLElement
+> &
+TextProps, "type">;
+
+type CheckboxInputFieldProps = Omit<FieldProps<
+  boolean,
+  FieldRenderProps<boolean, HTMLElement>,
+  HTMLElement
+> &
+  CheckboxInputProps, "type">;
+
+type EmailInputFieldProps = Omit<FieldProps<
+  string,
+  FieldRenderProps<string, HTMLElement>,
+  HTMLElement
+> &
+ EmailNumberProps, "type">;
+
+ type NewPasswordInputFieldProps = Omit<FieldProps<
+ string,
+ FieldRenderProps<string, HTMLElement>,
+ HTMLElement
+> &
+NewPasswordProps, "type">;
+
+type ExistingPasswordInputFieldProps = Omit<FieldProps<
+string,
+FieldRenderProps<string, HTMLElement>,
+HTMLElement
+> &
+ExistingPasswordProps, "type">;
+
+type InternalFieldProps = 
+{inputFieldComponent: React.FunctionComponent} &
+TextInputFieldProps | CheckboxInputFieldProps | EmailInputFieldProps | NewPasswordInputFieldProps | ExistingPasswordInputFieldProps;
+
+const InputField: FunctionComponent<InternalFieldProps> = props => {
+  const {
+    afterSubmit,
+    allowNull = false,
+    beforeSubmit,
+    format,
+    formatOnBlur,
+    initialValue,
+    isEqual,
+    parse,
+    subscription,
+    type,
+    validate,
+    validateFields,
+    name,
+    inputFieldComponent,
+    ...inputProps
+  } = props;
+  return (
+    <Field
+      afterSubmit={afterSubmit}
+      allowNull={allowNull}
+      beforeSubmit={beforeSubmit}
+      format={format}
+      formatOnBlur={formatOnBlur}
+      initialValue={initialValue}
+      isEqual={isEqual}
+      parse={parse}
+      subscription={subscription}
+      type={type}
+      validate={validate}
+      validateFields={validateFields}
+      name={name}
+      render={({ input, meta }) => {
+        <inputFieldComponent 
+        {...input}
+          {...inputProps}
+          textType={type}
+          error={meta.touched && meta.error}
+      />
+      }}
+    />
+  );
+};
+
+// It's not necessary to pass the type prop for a Field
```

Comment:
I think it is a sensible requirement to require the user to define a "type" of Field. We could default it to "text". I would expect there to be a type defined for "FieldType" or something. Maybe something like,

```ts
type FieldType = "text" | "checkbox" | "email" | "password";
```

Then based on the value of this prop we could constrain the props and render the appropriate field.

Reply:
We has discussed a better way of returning the correct inputs (with their specified props) in the Field component, but I haven't found a better way yet. I recall Button may have used some of this logic - still have to look into that.

Reply:
I'm punting on this idea for now.